server/text_generation_server/models/paged_causal_lm.py

```diff
@@ -413,12 +417,17 @@ def _prefill(
         )
 
         t0 = time.time_ns()
-        output = self.model(
-            input_ids,
-            position_ids=position_ids,
-            cache_data=cache_data,
-            return_embeds=True,
-        )
+        try:
+            output = self.model(
+                input_ids,
+                position_ids=position_ids,
+                cache_data=cache_data,
+                return_embeds=True,
+            )
+        except:
+            # if something goes wrong during forward, we still need to set the sequence ids
+            batch.sequence_ids = cache_data.sequence_ids
```

Comment:
Would it make sense to just move this from the bottom of the method to above the call to `self.model(...)`?

Reply:
I think `cache_data.sequence_ids` only gets populated within the call to `self.model(...)` so we can't move it beforehand.

Reply:
This feels a bit fragile, wonder if it would be better to revert to prior state (if possible) if the call to call to `self.model` fails? ideally within that call... i.e. avoid partial success.

Reply:
I agree that its fragile, and there might be a better way to address it from within the function. Not sure whether to prioritize that at this stage though.